### PR TITLE
Persist auth information to the iOS keychain

### PR DESCRIPTION
**Pull request by schnecle**
* Add persistence to keychain with error handling
* Add a wrapper around the keychain interactions to facilitate unit testing
* Add unit tests for FIRAppDistributionAuthPersistence

### Testing

  * Added unit tests for FIRAppDistributionAuthPersistence
  * Ran end to end locally to ensure that basic functionality works on device

### CODE REVIEW

File: FirebaseAppDistribution/Sources/FIRAppDistributionKeychainUtility.m [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#import <AppAuth/AppAuth.h>
+#import "FIRAppDistributionKeychainUtility+Private.h"
+
+NSString *const kFIRAppDistributionKeychainErrorDomain = @"com.firebase.app_distribution.keychain";
+
+@implementation FIRAppDistributionKeychainUtility
```
Comment (maksymmalyhin) [inline FirebaseAppDistribution/Sources/FIRAppDistributionKeychainUtility.m:19]: FYI, there is a generic [keychain storage](https://github.com/firebase/firebase-ios-sdk/blob/e3f3a78aefaafdab4fcfb41d24ccc93b7a5279e9/GoogleUtilities/Environment/Public/GULKeychainStorage.h#L24) introduced recently in GoogleUtilities which may help you make the code simpler. Feel free to reach out to me for usage details.

Comment (schnecle) [reply FirebaseAppDistribution/Sources/FIRAppDistributionKeychainUtility.m:19]: ooo! thank you for the tip! I will look into this - thanks!

File: FirebaseAppDistribution/Sources/FIRAppDistribution.m [hunk 2]
```diff
@@ -128,50 +136,70 @@ - (void)signInTesterWithCompletion:(void (^)(NSError *_Nullable error))completio
...
+      NSLog(@"Error fetching fresh tokens: %@", [error localizedDescription]);
+      [self signOutTester];
+      return;
+    }
+
+    // perform your API request using the tokens
+    NSURLSession *URLSession = [NSURLSession sharedSession];
+    NSMutableURLRequest *request = [[NSMutableURLRequest alloc] init];
+    NSString *URLString =
+        [NSString stringWithFormat:kReleasesEndpointURL, [[FIRApp defaultApp] options].googleAppID];
```
Comment (samedson) [inline FirebaseAppDistribution/Sources/FIRAppDistribution.m:165]: Recommend grabbing and holding on to the `googleAppID` in your SDK's initializer `initWithApp`. Seems to be best practice and I believe defaultApps can be deleted.

File: FirebaseAppDistribution/Sources/FIRAppDistributionKeychainUtility.m (hunk 1 above)
Comment (pranavrajgopal) [reply FirebaseAppDistribution/Sources/FIRAppDistributionKeychainUtility.m:19]: ooh this is awesome! Thanks @maksymmalyhin

Comment (schnecle) [reply FirebaseAppDistribution/Sources/FIRAppDistributionKeychainUtility.m:19]: @maksymmalyhin would you be alright with merging this PR as is and following up in a subsequent change?

Comment (maksymmalyhin) [reply FirebaseAppDistribution/Sources/FIRAppDistributionKeychainUtility.m:19]: Sure, it is definitely not a blocker to me. Just wanted to share the lib that may be useful for you.

File: FirebaseAppDistribution/Sources/FIRAppDistribution.m (hunk 2 above)
Comment (schnecle) [reply FirebaseAppDistribution/Sources/FIRAppDistribution.m:165]: We're going to address this in a future PR, as it is not actually related to the change for auth persistence.

**Pull request merged**